interface/backend/submission.py

```diff
@@ -1,18 +1,27 @@
 import logging
+from zipfile import ZipFile
 
 from django.db import transaction
 from django.utils import timezone
 
-import interface.backend.minio_api as storage
 from interface.models import Assignment
+import interface.backend.minio_api as storage
 
 
 log = logging.getLogger(__name__)
 
 
+class CorruptZipFile(Exception):
+    pass
+
+
 def handle_submission(file, assignment, user):
     log.debug(f'Submission {file.name} received')
 
+    with ZipFile(file) as archive:
+        if archive.testzip():
+            raise CorruptZipFile()
```

Comment:
Q: Can't we raise the same exception ```zipfile.BadZipFile```?

Reply:
A: I want to have a clear distinction between 'Hey, you sent a txt file instead of a valid zip archive' and 'Hey, you sent a zip, but it is corrupt'. Because `ZipFile(file)` raises the exception `BadZipFile` and the function `testzip` just returns the name of the first file that has a bad checksum I wanted to integrate it nicely with the already existing `try catch` clause.

Reply:
Yep, then it is ok :)